project/src/js/ui/inbox/thread/threadView.tsx

```diff
@@ -0,0 +1,91 @@
+import layout from '../../layout';
+import i18n from '../../../i18n';
+import redraw from '../../../utils/redraw';
+import { ThreadState, Post, ThreadAttrs } from '../interfaces';
+
+export default function view(vnode: Mithril.Vnode<ThreadAttrs>) {
+  const ctrl = vnode.state as ThreadState;
+  const headerCtrl = () => headerWidget(null,
+    backButton(ctrl.thread() ? ctrl.thread().name : null)
+  );
+  const bodyCtrl = threadBody.bind(undefined, ctrl);
+  window.addEventListener('native.keyboardshow', () => { (document.activeElement as any).scrollIntoView(true) });
+  return layout.free(headerCtrl, bodyCtrl, undefined, undefined);
+}
+
+function threadBody(ctrl: ThreadState) {
+  if (!ctrl.thread()) return null;
+  return (
+    <div key={ctrl.thread().id} className="threadWrapper native_scroller">
+      {ctrl.thread().posts.map(renderPost)}
+      <div className="responseWrapper">
+        <form id="responseForm"
+        onsubmit={function(e: Event) {
+          e.preventDefault();
+          const form = e.target as HTMLFormElement;
+          const body = (form[1] as HTMLTextAreaElement).value;
+          if (body.length >= 3)
+            return ctrl.sendResponse(form);
+          else
+            window.plugins.toast.show('Minimum length is 3', 'short', 'center');
+        }}>
+          <input id="id" value={ctrl.id()} type="hidden" />
+          <textarea id="body" className="responseBody" />
+          <button key="send" className="fatButton sendResponse" oncreate={h.autofocus} type="submit">
+            <span className="fa fa-check" />
+            {i18n('send')}
+          </button>
+          { deleteButton (ctrl) }
+        </form>
+      </div>
+    </div>
+  );
+}
+
+function renderPost(post: Post, index: number, posts: Array<Post>) {
+  let postClass = 'postWrapper';
+  if (index === 0)
+    postClass += ' first';
+  if (index === posts.length-1)
+    postClass += ' last';
+  return (
+    <div className={postClass} key={post.createdAt}>
+      <div className="infos">
+        {userStatus(post.sender)}
+        <span className="arrow" data-icon="H" />
+        {userStatus(post.receiver)}
+        &nbsp;–&nbsp;
+        {postDateFormat(post.createdAt)}
+      </div>
+      <div className="text">{post.text}</div>
+    </div>
+  );
+}
+
+function postDateFormat (timeInMillis: number) {
+  const time = window.moment(timeInMillis);
+  return time.format('MMM D H:mm');
+}
+
+function deleteButton (ctrl: ThreadState) {
+  return ctrl.deleteAttempted() ? (
+    <div className="negotiation confirmDeleteDialog">
+      <button key="confirmDelete" className="fatButton confirmDelete" oncreate={h.ontapY(() => {ctrl.deleteThread(ctrl.id());})}>
+        <span className="fa fa-trash-o" />
+        Delete
+      </button>
+      <button key="cancelDelete" className="fatButton cancelDelete" oncreate={h.ontapY(() => {ctrl.deleteAttempted(false); redraw();})}>
```

Comment:
I don't like onliners with two statements separated by semicolon. It's better like this:
```
<button key="cancelDelete" className="fatButton etc"
  oncreate={h.ontapY(() => {
    // do stuff
    // several lines  
 })}
/>
...
```